src/design_notes/auto_traits.md

```diff
@@ -0,0 +1,32 @@
+# Auto traits
+
+Auto traits permit automatically implementing a trait for types which contain
+fields implementing the trait. That is, they are fairly close to an automatic
+derive. They describe properties of types rather than behaviors; current stable
+Rust has two auto traits: `Send` and `Sync`.
```

Comment:
+ `Unpin` + `UnwindSafe`

Reply:
```
$ rg 'auto trait' library/
library/core/src/marker.rs
39:pub unsafe auto trait Send {
470:pub unsafe auto trait Sync {
717:pub(crate) unsafe auto trait Freeze {}
773:pub auto trait Unpin {}

library/rtstartup/rsbegin.rs
25:auto trait Sync {}
29:auto trait Freeze {}

library/rtstartup/rsend.rs
17:auto trait Freeze {}

library/std/src/panic.rs
107:/// panic). This trait is an auto trait, so it is automatically implemented for
141:pub auto trait UnwindSafe {}
159:pub auto trait RefUnwindSafe {}
```

Reply:
Freeze is an "impl detail" (but perhaps worth mentioning)